micro-rdk-installer/src/main.rs

```diff
@@ -214,13 +239,18 @@ fn write_credentials_to_app_binary(
     Ok(())
 }
 
-fn flash(args: WriteFlashArgs, config: &Config, app_path: PathBuf) -> Result<(), Error> {
+fn flash(
+    flash_args: FlashArgs,
+    monitor_args: Option<MonitorArgs>,
+    config: &Config,
+    app_path: PathBuf,
+) -> Result<(), Error> {
     log::info!("Connecting...");
     let mut flasher = connect(
-        &args.monitor_args.connect_args,
+        &monitor_args.clone().unwrap().connect_args,
```

Comment:
I see we're assuming `monitor_args` is always some `flash_args.monitor` is true. Instead of assuming, could we check that we're in a valid state regarding these two settings (and return an error if we're not)?

Reply:
actually, it might not be necessary anymore for `MonitorArgs` to be `Option`. checking now